src/coreclr/jit/codegenxarch.cpp

```diff
@@ -2010,8 +2010,6 @@ void CodeGen::genAllocLclFrame(unsigned frameSize, regNumber initReg, bool* pIni
         GetEmitter()->emitIns_R_AR(INS_lea, EA_PTRSIZE, REG_STACK_PROBE_HELPER_ARG, REG_SPBASE, spOffset);
         regSet.verifyRegUsed(REG_STACK_PROBE_HELPER_ARG);
 
-        // Can't have a call until we have enough padding for ReJit.
-        genPrologPadForReJit();
```

Comment:
Is `JIT_FLAG_PROF_REJIT_NOPS` still going to work with this change?

`JIT_FLAG_PROF_REJIT_NOPS` is not used currently. It may be best to delete it instead of keeping partially broken implementation around.

Reply:
> JIT_FLAG_PROF_REJIT_NOPS is not used currently. It may be best to delete it instead of keeping partially broken implementation around.

@jkotas Would it be fine with you if I undo the change to the jit file in this PR and come back with another PR removing `JIT_FLAG_PROF_REJIT_NOPS` flag and all the related infrastructure?

Reply:
Fine with me.

Reply:
Opened #46560 to address this